File: sample-projects/ruby/lib/de/sots/cellars_and_centaurs/domain/model/armor_class.rb
# frozen_string_literal: true

require_relative '../../version'
require_relative '../../application'

module De::Sots::CellarsAndCentaurs::Domain::Model
  class ArmorClass
    attr_accessor :description, :base, :bonus, :total

    def initialize(base, bonus, description = De::Sots::CellarsAndCentaurs::Application::CreatureUtil::STANDARD_ARMOR_CLASS_DESCRIPTION)
      @description = description
      @base = base
      @bonus = bonus
      @total = base + bonus
    end

    def natural?
      description == De::Sots::CellarsAndCentaurs::Application::CreatureUtil::STANDARD_ARMOR_CLASS_DESCRIPTION
    end
  end
end

File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/csharp/src/de/sots/cellarsandcentaurs/domain/model/ArmorClass.cs
using de.sots.cellarsandcentaurs.application;

namespace de.sots.cellarsandcentaurs.domain.model
{
    public class ArmorClass
    {
        private int value;
        private string description;

        public ArmorClass(int value)
        {
            this.value = value;
            this.description = CreatureUtil.STANDARD_ARMOR_CLASS_DESCRIPTION;
        }

        public int GetValue()
        {
            return value;
        }

        public void SetValue(int value)
        {
            this.value = value;
        }
    }
}
